src/primitives/hrp.rs

```diff
@@ -538,4 +614,45 @@ mod tests {
         let hrp = Hrp::parse_unchecked(s);
         assert_eq!(hrp.as_bytes(), s.as_bytes());
     }
+
+    #[test]
+    fn parse_display() {
+        let hrp = Hrp::parse_display(format_args!("{}_{}", 123, "abc")).unwrap();
+        assert_eq!(hrp.as_str(), "123_abc");
+
+        let hrp = Hrp::parse_display(format_args!("{:083}", 1)).unwrap();
+        assert_eq!(
+            hrp.as_str(),
+            "00000000000000000000000000000000000000000000000000000000000000000000000000000000001"
+        );
+
+        assert_eq!(Hrp::parse_display(format_args!("{:084}", 1)), Err(Error::TooLong(84)),);
+
+        assert_eq!(
+            Hrp::parse_display(format_args!("{:83}", 1)),
+            Err(Error::InvalidAsciiByte(b' ')),
+        );
+    }
+
+    #[test]
+    fn parse_non_ascii() {
+        assert_eq!(Hrp::parse("❤").unwrap_err(), Error::NonAsciiChar('❤'));
+    }
+
+    #[test]
+    fn parse_display_non_ascii() {
+        assert_eq!(Hrp::parse("❤").unwrap_err(), Error::NonAsciiChar('❤'));
```

Comment:
Should this call `parse_display` here?

Reply:
Oops, yes! Fixed.

Reply:
A bug in the unit tests, what are the odds ;)

Reply:
Oh actually, if this is based on the unit test code I posted above I intentionally included one that parsed using `Hrp::parse`.

Reply:
Sweet, cheers. Its hard to get good help :)